ssh/session.go

```diff
@@ -83,105 +83,162 @@ func handlePty(s *Session) {
 	}
 }
 
-func NewSession(target string, session sshserver.Session) (*Session, error) {
-	// Splits the target into a user, 0, and  the true target, 1.
-	// Example: username@shellhub.00-00-00-00-00-00@localhost.
+type Target struct {
+	Username string
+	Data     string
+}
+
+func NewTarget(target string) (*Target, error) {
+	// Target could be a either device id or a SSHID.
+	//
+	// Example: namespace.00-00-00-00-00-00
+	// 'namespace' is the user's namespace in ShellHub.
+	// '00-00-00-00-00' is the device's hostname in ShellHub.
+	//
+	// Example: username@namespace.00-00-00-00-00-00@localhost.
 	// 'username' is the user on the device.
-	// 'shellhub' is the user's namespace in ShellHub.
+	// 'namespace' is the user's namespace in ShellHub.
 	// '00-00-00-00-00' is the device's hostname in ShellHub.
+	// 'localhost' is the server's address.
+	const USERNAME = 0
+	const DATA = 1
+
 	parts := strings.SplitN(target, "@", 2)
 	if len(parts) != 2 {
-		return nil, ErrInvalidSessionTarget // Cloud not split user and the target to connect.
+		return nil, fmt.Errorf("cloud not split the target into two parts")
 	}
 
-	s := &Session{
-		session: session,
-		UID:     session.Context().Value(sshserver.ContextKeySessionID).(string),
-		User:    parts[0], // username.
-		Target:  parts[1], // shellhub.00-00-00-00-00-00@localhost.
+	return &Target{Username: parts[USERNAME], Data: parts[DATA]}, nil
+}
+
+// isSSHID checks if target is a SSHID.
+func (t *Target) isSSHID() bool {
+	return strings.Contains(t.Data, ".")
+}
+
+// isID checks if target is a ID.
+func (t *Target) isID() bool {
+	return !strings.Contains(t.Data, ".")
+}
```

Comment:
I'd drop this method to check if the target isn't an SSHID. So `!isSSHDID()` is equivalent.